staging/src/k8s.io/client-go/util/cert/cert.go

```diff
@@ -107,7 +107,13 @@ func GenerateSelfSignedCertKey(host string, alternateIPs []net.IP, alternateDNS
 // Certs/keys not existing in that directory are created.
 func GenerateSelfSignedCertKeyWithFixtures(host string, alternateIPs []net.IP, alternateDNS []string, fixtureDirectory string) ([]byte, []byte, error) {
```

Comment:
I realize that I'm critiquing the existing algorithm, but am I right that the self-signed in memory cert can have a 100 year lifespan if the fixture are missing?

I think this function could use a refactor.